src/toolbars/SnappingToolBar.cpp

```diff
@@ -0,0 +1,425 @@
+                  settings.newSnapMode != SnapMode::SNAP_OFF);
+
+            if (mSnapToCombo)
+               mSnapToCombo->Enable(settings.newSnapMode != SnapMode::SNAP_OFF);
+         }))
+{
+}
+
+SnappingToolBar::~SnappingToolBar()
+{
+}
+
+bool SnappingToolBar::ShownByDefault() const
+{
+   return true;
+}
+
+ToolBar::DockID SnappingToolBar::DefaultDockID() const
+{
+   return BotDockID;
+}
+
+SnappingToolBar& SnappingToolBar::Get(AudacityProject& project)
+{
+   auto &toolManager = ToolManager::Get( project );
+   return *static_cast<SnappingToolBar*>(toolManager.GetToolBar(ID()));
+}
+
+const SnappingToolBar& SnappingToolBar::Get(const AudacityProject& project)
+{
+   return Get( const_cast<AudacityProject&>( project )) ;
+}
+
+void SnappingToolBar::Create(wxWindow* parent)
+{
+   ToolBar::Create(parent);
+   UpdatePrefs();
+}
+
+void SnappingToolBar::Populate()
+{
+   SetBackgroundColour( theTheme.Colour( clrMedium  ) );
+
+   auto sizer = safenew wxFlexGridSizer(1, 1, 1);
+   Add(sizer, 0, wxALIGN_CENTER_VERTICAL | wxLEFT, 5);
+
+   mSnapModeCheckBox =
+      safenew wxCheckBox(this, wxID_ANY, XO("Snap").Translation());
+   mSnapModeCheckBox->SetName(XO("Snap").Translation());
+#if wxUSE_ACCESSIBILITY
+   // so that name can be set on a standard control
+   mSnapModeCheckBox->SetAccessible(
+      safenew WindowAccessible(mSnapModeCheckBox));
+#endif
+
+   sizer->Add(mSnapModeCheckBox, 0, wxBOTTOM | wxRIGHT | wxEXPAND, 5);
+
+   const bool snapEnabled =
+      ProjectSnap::Get(mProject).GetSnapMode() != SnapMode::SNAP_OFF;
+
+   mSnapModeCheckBox->SetValue(snapEnabled);
+
+   mSnapToCombo = safenew wxComboCtrl(
+      this, wxID_ANY, {}, wxDefaultPosition, wxDefaultSize /*, wxCB_READONLY*/);
+#if wxUSE_ACCESSIBILITY
+   // so that name can be set on a standard control
+   mSnapToCombo->SetAccessible(safenew WindowAccessible(mSnapToCombo));
+#endif
+
+   //mSnapToCombo->SetEditable(false);
+   mSnapToCombo->SetPopupControl(safenew SnapModePopup(mProject));
+   mSnapToCombo->SetName(mSnapToCombo->GetValue());
+   mSnapToCombo->Enable(snapEnabled);
+   mSnapToCombo->SetMinSize(wxSize(150, -1));
+   
+   sizer->Add(mSnapToCombo, 1, wxRIGHT | wxEXPAND, 5);
+
+   mSnapModeCheckBox->Bind(
+      wxEVT_CHECKBOX, [this](auto&) { OnSnapModeChanged(); });
+
+   mSnapModeCheckBox->Bind(
+      wxEVT_CHAR_HOOK,
+      [this](auto& evt)
+      {         
+         const auto keyCode = evt.GetKeyCode();
+
+         if (keyCode != WXK_NUMPAD_ENTER && keyCode != WXK_RETURN)
+         {
+            evt.Skip();
+            return;
+         }
+
+         mSnapModeCheckBox->SetValue(!mSnapModeCheckBox->GetValue());
+
+         OnSnapModeChanged();
+      });
+
+   RegenerateTooltips();
+   Fit();
+   Layout();
+}
+
+void SnappingToolBar::UpdatePrefs()
+{
+   // Set label to pull in language change
+   SetLabel(XO("Snapping"));
+
+   RegenerateTooltips();
+   // Give base class a chance
+   ToolBar::UpdatePrefs();
```

Comment:
Does it update for change of language?

Reply:
This needs to fixed, yes. Turns out that the TimeToolbar doesn't work either.

Reply:
I have rechecked, changing the language re creates all the toolbars from scratch. I thought that time toolbar was broken as I had untranslated Beats&Bars mode set

Reply:
I had forgot whether you needed special work with each toolbar to get translation update.  Seeing `UpdatePrefs` reminded me that language is one preference.